extensions/objects/templates/ImageWithRegionsEditor.js

```diff
@@ -300,10 +300,26 @@ oppia.directive('imageWithRegionsEditor', [
         $scope.getCursorStyle = function() {
           return ($scope.regionDrawMode) ? 'crosshair' : 'default';
         };
-
         $scope.resetEditor = function() {
-          $scope.$parent.value.imagePath = '';
-          $scope.$parent.value.labeledRegions = [];
+          $modal.open({
+            templateUrl: 'modals/clearImageRegions',
+            backdrop: true,
+            controller: [
+              '$scope', '$modalInstance', function($scope, $modalInstance) {
+                $scope.confirmClear = function() {
+                  $modalInstance.close();
+                  alertsService.clearWarnings();
+                };
+                $scope.cancel = function() {
+                  $modalInstance.dismiss('cancel');
+                  alertsService.clearWarnings();
+                };
+              }
+            ]
+          }).result.then(function() {
+            $scope.$parent.value.imagePath = '';
```

Comment:
As long as the expectations are correct, either behavior is fine. I would just make sure that the code is consistent with the text on the action button that the user clicks, and the text in the confirmation modal that pops up.